java-connectors/kafka-connect-gcp-pubsub/src/main/java/io/lenses/streamreactor/connect/gcp/pubsub/source/mapping/SourceRecordConverter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.gcp.pubsub.source.mapping;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.lenses.streamreactor.common.config.base.intf.Converter;
+import io.lenses.streamreactor.connect.gcp.pubsub.source.subscriber.PubSubMessageData;
+import lombok.AllArgsConstructor;
+
+/**
+ * SourceRecordConverter is responsible for converting PubSubMessageData to SourceRecord.
+ * It uses the MappingConfig to generate the key and value for the SourceRecord.
+ */
+@AllArgsConstructor
+public class SourceRecordConverter extends Converter<PubSubMessageData, SourceRecord> {
+
+  private final MappingConfig mappingConfig;
+
+  @Override
+  protected SourceRecord convert(final PubSubMessageData source) throws ConfigException {
+    return new SourceRecord(
+        source.getSourcePartition().toMap(),
+        source.getSourceOffset().toMap(),
+        source.getTargetTopicName(),
+        generateKeySchema(),
```

Comment:
Minor style related comment.

The generate*** sounds like a dynamic value calculated everytime, and my understanding is that it's fixed. Would it not be better to be get***. Is there value in creating the extra methods since they are one liner and private?

Reply:
I've changed it to `get***`.  I like the extra methods as it is more succinct and easier to read when scanning over the code.